### DISCUSSION TITLE
The download of the Snyk CLI was interrupted by a network error

### DISCUSSION THREAD

**Discussion by Zapzatron** [snyk/snyk-intellij-plugin#462] [Q&A]
![image](https://github.com/snyk/snyk-intellij-plugin/assets/100136305/cfa91bff-0609-45d2-9887-9de74b672a02)

**Comment by wolfswolke** [comment]
Same issue here for like more then a month. 
![image](https://github.com/snyk/snyk-intellij-plugin/assets/34097612/6c8ea5c8-8a3a-45c4-9377-38ba79bf94e7)
```2023-12-21 18:18:59,698 [  37187]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-12-21 18:18:59,810 [  37299]   INFO - #c.i.c.ComponentStoreImpl - Saving appProjectJdkTable took 13 ms
2023-12-21 18:19:27,730 [  65219]   INFO - #c.i.u.g.s.GistStorageImpl - Cleaning old huge-gists dirs from [C:\Users\zkwolf\AppData\Local\JetBrains\PyCharmCE2023.3\caches\huge-gists] ...
2023-12-21 18:19:27,731 [  65220]   INFO - #c.i.u.g.s.GistStorageImpl - Cleaning old huge-gists dirs finished
2023-12-21 18:20:16,165 [ 113654]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-12-21 18:20:16,186 [ 113675]   INFO - #c.i.c.ComponentStoreImpl - Saving appProjectJdkTable took 12 ms
2023-12-21 18:21:05,422 [ 162911]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-12-21 18:21:05,443 [ 162932]   INFO - #c.i.c.ComponentStoreImpl - Saving appProjectJdkTable took 13 ms
2023-12-21 18:21:25,924 [ 183413]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-12-21 18:21:25,948 [ 183437]   INFO - #c.i.c.ComponentStoreImpl - Saving appProjectJdkTable took 16 ms
2023-12-21 18:21:35,155 [ 192644]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-12-21 18:22:43,390 [ 260879]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-12-21 18:22:43,411 [ 260900]   INFO - #c.i.c.ComponentStoreImpl - Saving appProjectJdkTable took 15 ms
2023-12-21 18:22:57,540 [ 275029]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-12-21 18:23:12,687 [ 290176]   INFO - #c.i.w.i.i.j.s.JpsGlobalModelSynchronizerImpl - Saving global entities to files
2023-12-21 18:23:13,122 [ 290611]   WARN - #io.snyk.plugin.services.download.CliDownloaderErrorHandler - Retry of downloading the Snyk CLI failed.
java.nio.file.AccessDeniedException: C:\Users\zkwolf\AppData\Roaming\JetBrains\PyCharmCE2023.3\plugins\snyk-intellij-plugin\snyk-win.exe13876780125523633279.download -> C:\Users\zkwolf\AppData\Roaming\JetBrains\PyCharmCE2023.3\plugins\snyk-intellij-plugin\snyk-win.exe
	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:89)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:317)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:293)
	at java.base/java.nio.file.Files.move(Files.java:1432)
	at io.snyk.plugin.services.download.CliDownloader.downloadFile(CliDownloader.kt:58)
	at io.snyk.plugin.services.download.CliDownloaderErrorHandler$retryDownload$$inlined$runBackgroundableTask$1.run(progress.kt:39)
	at com.intellij.openapi.progress.impl.CoreProgressManager.startTask(CoreProgressManager.java:434)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.startTask(ProgressManagerImpl.java:132)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcessWithProgressAsynchronously$5(CoreProgressManager.java:484)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$4(ProgressRunner.java:248)
	at com.intellij.openapi.progress.ProgressManager.lambda$runProcess$0(ProgressManager.java:73)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$runProcess$1(CoreProgressManager.java:192)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:610)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:685)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:641)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:609)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:78)
	at com.intellij.openapi.progress.impl.CoreProgressManager.runProcess(CoreProgressManager.java:179)
	at com.intellij.openapi.progress.ProgressManager.runProcess(ProgressManager.java:73)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$submit$5(ProgressRunner.java:248)
	at com.intellij.openapi.progress.impl.ProgressRunner$ProgressRunnable.run(ProgressRunner.java:498)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:81)
	at com.intellij.util.concurrency.ChildContext$runAsCoroutine$1.invoke(propagation.kt:81)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:86)
	at com.intellij.util.concurrency.ChildContext.runAsCoroutine(propagation.kt:81)
	at com.intellij.openapi.progress.impl.ProgressRunner.lambda$launchTask$18(ProgressRunner.java:464)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:702)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:699)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:699)
	at java.base/java.lang.Thread.run(Thread.java:840)
2023-12-21 18:23:13,124 [ 290613]   WARN - #io.snyk.plugin.ui.SnykBalloonNotifications - The download of the Snyk CLI was interrupted by a network error (C:\Users\zkwolf\AppData\Roaming\JetBrains\PyCharmCE2023.3\plugins\snyk-intellij-plugin\snyk-win.exe7047897120578988236.download -> C:\Users\zkwolf\AppData\Roaming\JetBrains\PyCharmCE2023.3\plugins\snyk-intellij-plugin\snyk-win.exe). Do you want to try again?```

**Comment by Alatyami** [comment]
Run As Administrator doesn't solve it. Still looking for a solution.

**Comment by billbarni** [comment]
Months gone by, still not working. Several plugin updates already, running the IDE as admin... nothing.

**Comment by Darpholgshon** [comment]
To workaround the issue.

In Settings -> Tools -> Snyk:  I unchecked the **Automatically Manage needed binaries**, and downloaded the binary manually from the releases doc:

https://static.snyk.io/cli/v1.666.0/release.json

which on Windows is - https://static.snyk.io/cli/v1.666.0/snyk-win.exe
then just overwrite the **snyk-win.exe** in the IntelliJ plugins folder.

I have also tried changing the URL to point at the exe directly and leave the box checked, This didn't appear to give any indication of doing anything so either it worked and didn't log, or it didn't do anything.

![image](https://github.com/snyk/snyk-intellij-plugin/assets/1083903/244e63b4-d162-49b2-a0d3-58b895a0fb95)

Basically according to the documentation if you uncheck the checkbox it will use whatever binary is at the end of the supplied path, and you are responsible for manually updating it. It solves the annoying error at the very least.

Whatever you do I hope this helps.

**Comment by billbarni** [comment]
Problem gets updated and fixed, then new version comes... and new cli installation is needed. Please fix this.

**Comment by matbarni** [comment]
Again, another plugin update and another regression. Problem persists. I work for a company that pays for Snyk enterprise and this is a comical error at this point.

**Comment by sudislife** [comment]
If you're using a package manager such as scoop, what worked for me was

`scoop install snyk` 
or
`scoop update snyk`

https://docs.snyk.io/developer-tools/snyk-cli/install-or-update-the-snyk-cli

Refer to docs for other OS

**Comment by SystemNatalie** [comment]
Experiencing the same issue in 2026.

`The download of the Snyk CLI was interrupted by an error (C:\Users\<REDACTED>\AppData\Roaming\JetBrains\IntelliJIdea2025.2\plugins\snyk-intellij-plugin\snyk-win.exe72663801751438495.download -> C:\Users\<REDACTED>\AppData\Roaming\JetBrains\IntelliJIdea2025.2\plugins\snyk-intellij-plugin\snyk-win.exe). Do you want to try again?`

**Comment by dgo42** [comment]
Almost permanent at each Snyk update

The download of the Snyk CLI was interrupted by an error (C:\Users\<SKIPPED>\AppData\Roaming\JetBrains\IntelliJIdea2026.1\plugins\snyk-intellij-plugin\snyk-win.exe11658775923349784574.download -> C:\Users\<SKIPPED>\AppData\Roaming\JetBrains\IntelliJIdea2026.1\plugins\snyk-intellij-plugin\snyk-win.exe). Do you want to try again?
